From 56851ea1ac0c16cc54a6c7f966e81aab27e9c6a7 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Fri, 31 Jul 2020 16:04:45 +0700
Subject: [PATCH 401/476] ARM: dts: r8a7745: Support VSP control by DU

This patch make changes to vsp device nodes so they can be
controlled by DU and VSPM kernel module.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/r8a7745.dtsi | 30 ++++++++++++++----------------
 1 file changed, 14 insertions(+), 16 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7745.dtsi b/arch/arm/boot/dts/r8a7745.dtsi
index aa152f0..abb3f0d 100644
--- a/arch/arm/boot/dts/r8a7745.dtsi
+++ b/arch/arm/boot/dts/r8a7745.dtsi
@@ -1424,32 +1424,20 @@
 			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
-		vsp@fe928000 {
-			compatible = "renesas,vsp1";
+		vspm@fe928000 {
+			compatible = "renesas,vspm-vsps", "renesas-vspm";
 			reg = <0 0xfe928000 0 0x8000>;
 			interrupts = <GIC_SPI 267 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1_SY>;
 			power-domains = <&cpg_clocks>;
-
-			renesas,has-lut;
-			renesas,has-sru;
-			renesas,#rpf = <5>;
-			renesas,#uds = <1>;
-			renesas,#wpf = <4>;
 		};
 
-		vsp@fe930000 {
-			compatible = "renesas,vsp1";
+		vspm@fe930000 {
+			compatible = "renesas,vspm-vspd0", "renesas-vspm";
 			reg = <0 0xfe930000 0 0x8000>;
 			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
 			clocks = <&mstp1_clks R8A7745_CLK_VSP1DU0>;
 			power-domains = <&cpg_clocks>;
-
-			renesas,has-lif;
-			renesas,has-lut;
-			renesas,#rpf = <4>;
-			renesas,#uds = <1>;
-			renesas,#wpf = <1>;
 		};
 
 		fdp0: fdpm@fe940000 {
@@ -1460,6 +1448,14 @@
 			power-domains = <&sysc R8A7745_PD_ALWAYS_ON>;
 		};
 
+		vspdu0: vspdu0@fe930000 {
+			compatible = "renesas,vsp2";
+			reg = <0 0xfe930000 0 0x7404>;
+			interrupts = <0 246 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mstp1_clks R8A7745_CLK_VSP1DU0>;
+			power-domains = <&cpg_clocks>;
+		};
+
 		du: display@feb00000 {
 			compatible = "renesas,du-r8a7745";
 			reg = <0 0xfeb00000 0 0x40000>;
@@ -1471,6 +1467,8 @@
 			clock-names = "du.0", "du.1";
 			status = "disabled";
 
+			vsps = <&vspdu0>;
+
 			ports {
 				#address-cells = <1>;
 				#size-cells = <0>;
-- 
2.7.4

